Update README.md with comprehensive documentation (#10)
IgniteUI/igniteui-grid-lite: README.md

> # Ignite UI Grid Lite for Web Components

Reviewer: ```suggestion
# Ignite UI Grid Lite
```

[![Coverage Status](https://coveralls.io/repos/github/IgniteUI/igniteui-grid-lite/badge.svg?branch=master)](https://coveralls.io/github/IgniteUI/igniteui-grid-lite?branch=master)
[![npm version](https://badge.fury.io/js/igniteui-grid-lite.svg)](https://badge.fury.io/js/igniteui-grid-lite)

Ignite UI Grid Lite is a high-performance, lightweight data grid built as a web component using [Lit](https://lit.dev/). It provides essential grid functionality with a small footprint, making it perfect for applications that need fast, efficient data visualization without the overhead of a full-featured grid.

## Table of Contents
- [Features](#features)
- [Browser Support](#browser-support)
- [Getting Started](#getting-started)
- [Usage](#usage)
- [Building the Library](#building-the-library)
- [Running Tests](#running-tests)
- [Documentation](#documentation)
- [Ignite UI Grid Lite vs. Ignite UI for Angular](#ignite-ui-grid-lite-vs-ignite-ui-for-angular)
- [Contributing](#contributing)
- [Support](#support)
- [License](#license)

## Features

Ignite UI Grid Lite provides core data grid capabilities with an emphasis on performance and simplicity:

> - **High Performance Virtualization** - Handles large datasets efficiently with built-in virtual scrolling

Reviewer 2: ```suggestion
- **High Performance Virtualization** - Handles large datasets efficiently with built-in virtual scrolling
```
The lite Grid has only Row Virtualization

- **Filtering** - Built-in column filtering capabilities
- **Sorting** - Column sorting support for better data organization
- **Lightweight** - Minimal bundle size for fast load times
- **Web Standards** - Built with web components for framework-agnostic usage
- **Modern Architecture** - Built with Lit and TypeScript for maintainability
- **Customizable** - Flexible theming and styling options

> - **Accessible** - Designed with accessibility in mind

Reviewer 2: add "Cell Templating" and "Header Templating"

npm install
```

### Build

To build the library:

```bash
npm run build
```

This will:
- Analyze custom elements
- Build the TypeScript source
- Generate API documentation

### Development

Start the development server with live reload:

```bash
npm start
```

This will open the demo page at `demo/index.html` with automatic rebuilding on file changes.

## Running Tests

Run tests once:

```bash
npm test
```

Run tests in watch mode:

```bash
npm run test:watch
```

Tests are written using [@web/test-runner](https://modern-web.dev/docs/test-runner/overview/) and [@open-wc/testing](https://open-wc.org/docs/testing/helpers/).

## Documentation

### API Documentation

The API documentation is generated using [TypeDoc](https://typedoc.org/). To build the docs:

```bash
npm run build
```

Documentation will be available in the build output.

### Additional Resources

- [Ignite UI Product Documentation](https://www.infragistics.com/products/ignite-ui-angular)
- [Grid Lite Documentation](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid-lite/overview)
- [Web Components Guide](https://lit.dev/)

## Ignite UI Grid Lite vs. Ignite UI for Angular

> Ignite UI Grid Lite is designed as a lightweight, open-source alternative to the full-featured [Ignite UI for Angular Grid](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/overview).

Reviewer 2: ```suggestion
Ignite UI Grid Lite is designed as a lightweight, open-source alternative to the full-featured [Ignite UI for Web Components Data Grid](https://www.infragistics.com/products/ignite-ui-web-components/web-components/components/grids/data-grid).
```

### Ignite UI Grid Lite (Open Source - MIT)

**Best for:**
- Applications requiring basic grid functionality
- Projects where bundle size is critical
- Framework-agnostic web applications
- Community-driven development

**Includes:**
- Core data virtualization
- Basic filtering and sorting
- Lightweight and fast
- MIT licensed and free for all use

### Ignite UI for Angular (Commercial)

**Best for:**
- Enterprise applications requiring advanced features
- Complex data scenarios with editing, grouping, and aggregation
- Applications needing professional support and SLA

> - Angular-specific optimizations

Reviewer 2: ```suggestion

```
